lib/logstash/filters/lookup.rb

```diff
@@ -0,0 +1,183 @@
+  # http://localhost:8080/geoPoints/json
+  # http://localhost:8080/geoPoints/csv
+  # If no suffix matches, defaults to YAML
+  config :map_url, :validate => :string, :required => true
+
+  # When using a map file or url, this setting will indicate how frequently
+  # (in seconds) logstash will check the YAML file or url for updates.
+  config :refresh_interval, :validate => :number, :default => 300
+
+  # The destination field you wish to populate with the mapd code. The default
+  # is a field named `mapping`. Set this to the same value as source if you want
+  # to do a substitution, in this case filter will allways succeed. This will clobber
+  # the old value of the source field! 
+  config :destination, :validate => :string, :default => "mapping"
+
+  # In case no mapping occurs in the event (no matches), this will add a default
+  # mapping string, which will always populate `field`, if the match failed.
+  #
+  # For example, if we have configured `fallback => "no match"`, using this map:
+  # [source,ruby]
+  #     foo: bar
+  #
+  # Then, if logstash received an event with the field `foo` set to `bar`, the destination
+  # field would be set to `bar`. However, if logstash received an event with `foo` set to `nope`,
+  # then the destination field would still be populated, but with the value of `no match`.
+  # This configuration can be dynamic and include parts of the event using the `%{field}` syntax.
+  config :fallback, :validate => :string
+
+  def get_map
+    @my_map
+  end
+
+  def set_map(map)
+    @my_map = map;
+  end
+
+  public
+  def register
+    @my_map = {}
+    @next_refresh = Time.now + @refresh_interval
+    download_ws(@map_url, true)
+    @logger.debug? and @logger.debug("#{self.class.name}: map - ", :map => get_map)
+    type = 'Exact'
+    @logger.debug? and @logger.debug("#{self.class.name}: map mapping method - "+type)
+  end
+
+  # def register
+
+  def json_loader(data)
+    get_map.merge!(JSON.parse(File.read(data)))
+  end
+
+  def csv_loader(data)
+    data = CSV.read(data).inject(Hash.new) do |acc, v|
+      acc[v[0]] = v[1]
+      acc
+    end
+    get_map.merge!(data)
+  end
+
+  def yml_loader(data)
+    get_map.merge!(YAML.load_file(data))
+  end
+
+  def load_file(registering, extension, data)
+    begin
+      if extension.equal?('.json')
+        return json_loader(data)
+      elsif extension.end_with?('.csv')
+        return csv_loader(data)
+      end
+      yml_loader(data)
+    rescue Exception => _
+      if registering
+        raise "#{self.class.name}: Bad Syntax in map file #{file_name}"
+      else
+        @logger.warn("#{self.class.name}: Bad Syntax in map file, continuing with old map", :map_path => file_name)
+      end
+    end
+  end
+
+  def get_extension(path)
+    if path.end_with?('json')
+      return '.json'
+    elsif path.end_with?('csv')
+      return '.csv'
+    end
+    '.yml'
+  end
+
+  public
+  def download_ws(path, registering=false)
```

Comment:
Any reason why you save the response blob on disk and then reload it memory? Can't the parsing be done automatically from the response stream?